providers/os/resources/npm/npm.go

```diff
@@ -4,11 +4,53 @@
 package npm
 
 import (
+	"github.com/package-url/packageurl-go"
+	"go.mondoo.com/cnquery/v10/providers/os/resources/cpe"
 	"io"
-
-	"go.mondoo.com/cnquery/v10/providers-sdk/v1/upstream/mvd"
+	"strings"
 )
 
 type Parser interface {
-	Parse(r io.Reader) ([]*mvd.Package, error)
+	Parse(r io.Reader) (*Package, []*Package, error)
+}
+
+type Package struct {
+	Name        string
+	File        string
+	License     string
+	Description string
+	Version     string
+	Purl        string
+	Cpes        []string
+}
+
+// NewPackageUrl creates a npm package url for a given package name and version
+// see https://github.com/package-url/purl-spec/blob/master/PURL-TYPES.rst#npm
+func NewPackageUrl(name string, version string) string {
+	namespace := ""
+	// ensure the name is according to the PURL spec
+	name = strings.ReplaceAll(name, "_", "-")
+
+	components := strings.Split(name, "/")
+	if len(components) > 1 {
+		namespace = components[0]
+		name = components[1]
+	}
+
+	return packageurl.NewPackageURL(
+		packageurl.TypeNPM,
+		namespace,
+		name,
+		version,
+		nil,
+		"").String()
+}
+
+func NewCpes(name string, version string) []string {
+	cpes := []string{}
+	cpeEntry, err := cpe.NewPackage2Cpe(name, name, version, "", "")
+	if err == nil && cpeEntry != "" {
```

Comment:
can we have a case where there is no err but cpeEntry is an empty string? if not, we should make sure that the `NewPackage2Cpe` function returns an err

Reply:
I agree that we should handle errors. For that specific case I would like to avoid it to ensure that the cpe generation is to accidentally stopping a package gathering. To compensate, I added more tests for the cpe generation. I add comment to make this more clear.